subscription_mass_mailing/models/mailing_criterium.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+#   Houssine Bakkali <houssine@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+from datetime import date, datetime, timedelta  # noqa
+from odoo.tools import DEFAULT_SERVER_DATE_FORMAT as DTF
+# from odoo.tools.safe_eval import safe_eval
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+# operations available for python expression
+def pd(dt):
+    """parse datetime"""
+    return datetime.strptime(dt, DTF) if dt else dt
+
+
+def fd(dt):
+    """format date"""
+    return dt.strftime(DTF)
+
+
+def today_str():
+    return fields.Date.today()
+
+
+def today():
+    return pd(fields.Date.today())
+
+
+class MailingCriterium(models.Model):
+    _name = "ps.mailing.criterium"
+
+    name = fields.Char()
+    active = fields.Boolean(string="Active", default=True)
+    subscription = fields.Many2one(
+        comodel_name="subscription",
+        string="Subscription Template",
+        required=True,
+    )
+    mail_template = fields.Many2one(
+        comodel_name="mail.template",
+        string="Email Template",
+        required=True,
+    )
+    py_expr_filter = fields.Char(
+        string="Python Expression Filter",
+        required=False,
+        help="Available variables:\n"
```

Comment:
Trailing `\n` would be unnecessary if you used triple-quoted comments.

Reply:
I toke it like it was it the product_subscription_mass_mailing. I won't change it for now.